cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll

```diff
@@ -73,20 +73,13 @@ private int isSource(Expr bufferExpr, Element why) {
   )
 }
 
-private predicate localFlowToExprStep(DataFlow::Node n1, DataFlow::Node n2) {
-  not exists(n2.asExpr()) and
-  DataFlow::localFlowStep(n1, n2)
-}
-
-/** Holds if `n2 + delta` may be equal to `n1`. */
+/**
+ * Holds if data flow steps from `e1` to `e2` without stepping through any
+ * other intermediate expressions.
+ */
 private predicate localFlowStepToExpr(Expr e1, Expr e2) {
   getBufferSizeCand0(e1) and
```

Comment:
Nitpicking: since we only use this in one place, do we just want to inline this?

Reply:
Good idea. Fixed in 2617e6d7c6d1c4e38b81ba275f09c398c66ec8d2.